prestashop1.6/upgrade/Upgrade-1.2.0.php

```diff
@@ -0,0 +1,139 @@
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to support@tawk.to so we can send you a copy immediately.
+ *
+ * @author tawkto support@tawk.to
+ * @copyright Copyright (c) 2014-2021 tawk.to
+ * @license   http://opensource.org/licenses/osl-3.0.php  Open Software License (OSL 3.0)
+ */
+
+if (!defined('_PS_VERSION_')) {
+    exit;
+}
+
+function upgrade_module_1_2_0()
+{
+    $db = Db::getInstance();
+
+    // remove the suffix first on the config key names.
+    $remove_suffix_result = remove_suffix($db);
+    if (!$remove_suffix_result) {
+        return false;
+    }
+
+    // force reload cache
+    Configuration::loadConfiguration();
+
+    // insert the new records for TAWKTO_WIDGET_CURRENT.
+    $insert_records_result = insert_records();
+    if (!$insert_records_result) {
+        return false;
+    }
+
+    // delete records for TAWKTO_WIDGET_PAGE_ID & TAWKTO_WIDGET_WIDGET_ID
+    $remove_extras_result = remove_extras();
+    if (!$remove_extras_result) {
+        return false;
+    }
+
+    return true;
+}
+
+function remove_suffix($db)
+{
+    $keys = array(
+        TawkTo::TAWKTO_WIDGET_PAGE_ID,
+        TawkTo::TAWKTO_WIDGET_WIDGET_ID,
+        TawkTo::TAWKTO_WIDGET_OPTS,
+        TawkTo::TAWKTO_WIDGET_USER
+    );
+
+    // start building the update sql statement
+    $sql = array(
+        'UPDATE '._DB_PREFIX_.bqSQL(Configuration::$definition['table']).' conf',
+        'SET conf.name = CASE'
+    );
+
+    // build case when clause
+    foreach ($keys as $key) {
+        array_push($sql, "WHEN conf.name LIKE '".pSQL($key)."%' THEN '".pSQL($key)."'");
+    }
+
+    // build else and end case clause and where clause
+    array_push(
+        $sql,
+        "ELSE conf.name",
+        "END",
+        "WHERE conf.name LIKE 'TAWKTO_WIDGET_%';"
+    );
+
+    // join sql array and execute
+    $result = $db->execute(join(' ', $sql)); //returns boolean value
+
+    return $result;
+}
+
+function insert_records()
+{
+    $res = true;
+
+    // modify global first
+    $res &= modify_widget();
+
+    $shop_ids = Shop::getCompleteListOfShopsID();
+
+    $updated_groups = array();
+    foreach ($shop_ids as $shop_id) {
+        $shop_group_id = (int)Shop::getGroupFromShop($shop_id);
+
+        if (!in_array($shop_group_id, $updated_groups)) {
+            // update the group config
+            $res &= modify_widget($shop_group_id);
+            $updated_groups[] = $shop_group_id;
+        }
+
+        // update the shop config
+        $res &= modify_widget($shop_group_id, $shop_id);
+    }
+
+    return $res;
+}
+
+function modify_widget($shop_group_id = null, $shop_id = null)
+{
+    if (isset($shop_id)) {
+        Shop::setContext(Shop::CONTEXT_SHOP, $shop_id);
+    } elseif (isset($shop_group_id)) {
+        Shop::setContext(Shop::CONTEXT_GROUP, $shop_group_id);
+    } else {
+        Shop::setContext(Shop::CONTEXT_ALL);
+    }
```

Comment:
This is to compliment the issue above about the plugin being updated while a specific store was selected instead of `All Shops`. The `Configuration::updateValue` function was causing problems when inserting the records for `All Shops` and  `Shop Group`. And that is due to the function getting the current selected shop's `$shop_id` and `$shop_group_id` if they're `null`.